### ISSUE TITLE
Support PSUBSCRIBE

### ISSUE THREAD

**Issue by igorlesnenko** [davidyaha/graphql-redis-subscriptions#91]
Support PSUBSCRIBE

**Comment by davidyaha** [comment]
Hey @igorlesnenko!
Thanks for taking interest in this project!

Would you mind adding some more information regarding your use case?

I've actually played around with the idea of supporting PSUBSCRIBE but I do need a use case to follow and I couldn't think of a good one.

Thanks again,
David

**Comment by igorlesnenko** [comment]
@davidyaha 
For example, we have SaaS project management system where users can create Projects and Tasks. I push messages in channels like: `project.id123.task.id789`.

So If I want to subscribe on particular Task updates I will subscribe to `*.task.id789`
If I want to subscribe to all Tasks that belongs to project with `id=123` I will subscribe to `project.id123.task.*`.
If I want to subscribe to particular Project, I will subscribe to `project.id123`

What do you think?

I tried to just change subscribe to psubscribe, but looks like it is not that simple.

**Comment by igorlesnenko** [comment]
> I just tried to change subscribe to psubscribe, but looks like it is not that simple.

Ok, I also had to change onMessage resolver and now it works for me.

```js
this.redisSubscriber.on('pmessage', this.onMessage.bind(this));
```

```js
private onMessage(pattern: string, channel: string, message: string) {
    const subscribers = this.subsRefsMap[pattern];
...
}
```

**Comment by ivosabev** [comment]
+1 for pattern support

**Comment by davidyaha** [comment]
Hey @ivosabev! Care to continue where @j3ddesign left off?

**Comment by ntelkedzhiev** [comment]
@davidyaha Any intentions of adding this soon?

**Comment by davidyaha** [comment]
@ntelkedzhiev Available on version 2.1.0 thanks to @neilljordan
